egs/sad_rats/s5/local/get_speech_targets.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+
+# Copyright 2020  Desh Raj (Johns Hopkins University)
+# Copyright 2020  John Morgan (ARL)
+# Apache 2.0
+
+# This script prepares targets for whole recordings for training a Speech
+# Activity detector system.
+# It takes as input a data dir where it is assumed that MFCC features have
+# already been extracted.
+# It also takes an RTTM file  with speech segments.
+# It uses these segments to obtain per-frame targets for the recordings in the format:
+# [ silence speech ]
+
+from __future__ import division
+
+import argparse
+import logging
+import numpy as np
+import subprocess
+import sys
+import itertools
+from collections import defaultdict
+
+sys.path.insert(0, 'steps')
+import libs.common as common_lib
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        description="""This script prepares targets for whole recordings for training
+            an speech activity detector system. It just takes as input a data dir
+            where it is assumed that MFCC features have already been
+            extracted. It also takes an RTTM file containing
+            speech segments. It uses these segments to 
+            obtain per-frame targets for the recordings in the format:
+            [ silence speech ]
+        """)
+
+    parser.add_argument("--frame-shift", type=float, default=0.01,
+                        help="Frame shift value in seconds")
+    parser.add_argument("--label-smoothing", type=float, default=0.0,
+                        help="Value between 0 and 1. Amount of label smoothing to apply"
+                        "to get soft labels instead of one-hot labels")
+    parser.add_argument("reco2num_frames", type=str,
+                        help="""The number of frames per reco
+                        is used to determine the num-rows of the output matrix
+                        """)
+    parser.add_argument("rttm", type=str,
+                        help="Input RTTM file containing speech segments")
+    parser.add_argument("out_targets_ark", type=str,
+                        help="""Output archive to which the
+                        recording-level matrix will be written in text
+                        format""")
+
+    args = parser.parse_args()
+
+    if args.frame_shift < 0.0001 or args.frame_shift > 1:
+        raise ValueError("--frame-shift should be in [0.0001, 1]; got {0}"
+                         "".format(args.frame_shift))
+    return args
+
+class Segment:
```

Comment:
I believe you borrowed this from the `steps/overlap/get_overlap_targets.py`. Since the time I pushed that code, I have realized that it may be more "pythonic" to use a namedtuple here instead of this class. Perhaps something like the following:

```
from collections import namedtuple

Segment = namedtuple('Segment', 'reco spk start dur end')
```

And later on, in the for loop of line 103, you can do the following:

```
start = float(3)
duration = float(4)
end = start + duration
segments.append(Segment(fields[1], fields[7], start, duration, end))
```

This is better because each segment here is an immutable record, and so it is well represented by a namedtuple (which is immutable). We can also do the same through classes by using a `dataclass` as follows:

```
from dataclasses import dataclass

@dataclass(frozen=True)
class Segment:
    recoid: str,
    spkid: str,
    start: float,
    duration: float,
    end: float
```

But I think a `namedtuple` is more convenient here.

Reply:
I tried changing to namedtuple,but I ran into problems. I'm now reluctant to fix something that is not broken.

Reply:
Sorry, I made a small error in the code above. It should be: 

```
segments.append(
  Segment(
    reco = fields[1],
    spk = fields[7],
    start = start, 
    dur = duration, 
    end = end
))
```

i.e., you have to specify the key-value pairs when creating the namedtuple object

Reply:
I made the change you suggested and it looks like the namedtuple data structure works now.